ifcopenshell: 240611 -> 0.7.10, fix build and activate most tests

autra opened the pull request:
## Description of changes

Fix the ifcopenshell build.

Fix #312380 



## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

autra commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,15 +1,21 @@
 , opencollada
 , libxml2
+, hdf5
+, zlib
```
Is there a convention on the order here? (it's a bit random for now). Should I order alphabetically?

autra replied:
I ordered alphabetically.

autra commented:
Ok now it's compiling, but I have a weird problem with the runtime lib. libxml2 is missing at runtime:

```
 ❯ ldd result/bin/IfcConvert
        linux-vdso.so.1 (0x00007ffed5cc0000)
        libTKernel.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKernel.so.7 (0x00007f73733f4000)
        libTKMath.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKMath.so.7 (0x00007f7373000000)
        libTKBRep.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKBRep.so.7 (0x00007f73732c5000)
        libTKGeomBase.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKGeomBase.so.7 (0x00007f7372a00000)
        libTKGeomAlgo.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKGeomAlgo.so.7 (0x00007f7372400000)
        libTKG3d.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKG3d.so.7 (0x00007f7372ed8000)
        libTKG2d.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKG2d.so.7 (0x00007f73729a1000)
        libTKShHealing.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKShHealing.so.7 (0x00007f7372000000)
        libTKTopAlgo.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKTopAlgo.so.7 (0x00007f7371c00000)
        libTKMesh.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKMesh.so.7 (0x00007f73722ed000)
        libTKPrim.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKPrim.so.7 (0x00007f737293a000)
        libTKBool.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKBool.so.7 (0x00007f7371600000)
        libTKBO.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKBO.so.7 (0x00007f7371200000)
        libTKFillet.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKFillet.so.7 (0x00007f7370e00000)
        libTKSTEP.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKSTEP.so.7 (0x00007f7370a00000)
        libTKSTEPBase.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKSTEPBase.so.7 (0x00007f7370600000)
        libTKSTEPAttr.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKSTEPAttr.so.7 (0x00007f7371a39000)
        libTKXSBase.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKXSBase.so.7 (0x00007f7370200000)
        libTKSTEP209.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKSTEP209.so.7 (0x00007f7371f57000)
        libTKIGES.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKIGES.so.7 (0x00007f736fc00000)
        libTKOffset.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKOffset.so.7 (0x00007f737004c000)
        libTKHLR.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKHLR.so.7 (0x00007f73710d9000)
        libTKBin.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKBin.so.7 (0x00007f7373290000)
        libhdf5_cpp.so.310 => /nix/store/l34mr0cav6k6798gfx99hz02w5wwv1as-hdf5-cpp-1.14.3/lib/libhdf5_cpp.so.310 (0x00007f7371580000)
        libhdf5.so.310 => /nix/store/l34mr0cav6k6798gfx99hz02w5wwv1as-hdf5-cpp-1.14.3/lib/libhdf5.so.310 (0x00007f736f600000)
        libz.so.1 => /nix/store/zph9xw0drmq3rl2ik5slg0n2frw9lw5m-zlib-1.3.1/lib/libz.so.1 (0x00007f7372eba000)
        libboost_system.so.1.79.0 => /nix/store/mcj6hbf8amfq4ii6kqi510d43xgjq6as-boost-1.79.0/lib/libboost_system.so.1.79.0 (0x00007f7373289000)
        libboost_program_options.so.1.79.0 => /nix/store/mcj6hbf8amfq4ii6kqi510d43xgjq6as-boost-1.79.0/lib/libboost_program_options.so.1.79.0 (0x00007f7371eeb000)
        libboost_regex.so.1.79.0 => /nix/store/mcj6hbf8amfq4ii6kqi510d43xgjq6as-boost-1.79.0/lib/libboost_regex.so.1.79.0 (0x00007f737152f000)
        libboost_thread.so.1.79.0 => /nix/store/mcj6hbf8amfq4ii6kqi510d43xgjq6as-boost-1.79.0/lib/libboost_thread.so.1.79.0 (0x00007f737291c000)
        libboost_date_time.so.1.79.0 => /nix/store/mcj6hbf8amfq4ii6kqi510d43xgjq6as-boost-1.79.0/lib/libboost_date_time.so.1.79.0 (0x00007f7372917000)
        libxml2.so.2 => not found
        libpcre.so.1 => /nix/store/pq0cj8f364jvv16yqvkc43cz2vwcwg03-pcre-8.45/lib/libpcre.so.1 (0x00007f7370d86000)
        libstdc++.so.6 => /nix/store/c2yb135iv4maadia5f760b3xhbh6jh61-gcc-13.2.0-lib/lib/libstdc++.so.6 (0x00007f736f200000)
        libm.so.6 => /nix/store/ddwyrxif62r8n6xclvskjyy6szdhvj60-glibc-2.39-5/lib/libm.so.6 (0x00007f737091d000)
        libgcc_s.so.1 => /nix/store/c2yb135iv4maadia5f760b3xhbh6jh61-gcc-13.2.0-lib/lib/libgcc_s.so.1 (0x00007f73722c8000)
        libc.so.6 => /nix/store/ddwyrxif62r8n6xclvskjyy6szdhvj60-glibc-2.39-5/lib/libc.so.6 (0x00007f736f013000)
        /nix/store/ddwyrxif62r8n6xclvskjyy6szdhvj60-glibc-2.39-5/lib/ld-linux-x86-64.so.2 => /nix/store/anlf335xlh41yjhm114swi87406mq5pw-glibc-2.38-44/lib64/ld-linux-x86-64.so.2 (0x00007f73735e5000)
        libpthread.so.0 => /nix/store/ddwyrxif62r8n6xclvskjyy6szdhvj60-glibc-2.39-5/lib/libpthread.so.0 (0x00007f7372910000)
        librt.so.1 => /nix/store/ddwyrxif62r8n6xclvskjyy6szdhvj60-glibc-2.39-5/lib/librt.so.1 (0x00007f737290b000)
        libdl.so.2 => /nix/store/ddwyrxif62r8n6xclvskjyy6szdhvj60-glibc-2.39-5/lib/libdl.so.2 (0x00007f7372904000)
        libTKCAF.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKCAF.so.7 (0x00007f7370556000)
        libTKBinL.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKBinL.so.7 (0x00007f73708bf000)
        libTKLCAF.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKLCAF.so.7 (0x00007f736fb04000)
        libTKCDF.so.7 => /nix/store/p2y06mqk94i5sapgja08fvf9g5lzabxb-opencascade-occt-7.6.2/lib/libTKCDF.so.7 (0x00007f73704f7000)
        libicudata.so.73 => /nix/store/qwnmd1a9zxfjqgkhv3jyyf86r9jm4gsq-icu4c-73.2/lib/libicudata.so.73 (0x00007f736d000000)
        libicui18n.so.73 => /nix/store/qwnmd1a9zxfjqgkhv3jyyf86r9jm4gsq-icu4c-73.2/lib/libicui18n.so.73 (0x00007f736cc00000)
        libicuuc.so.73 => /nix/store/qwnmd1a9zxfjqgkhv3jyyf86r9jm4gsq-icu4c-73.2/lib/libicuuc.so.73 (0x00007f736c800000)
```
I have no idea how to fix this problem, for me `buildInputs` was supposed to do the job (and it does so for opencascade)... @fehnomenal  any idea?

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/fixing-ifcopenshell-libxml2-not-found-by-linker-at-runtime/45764/1

autra commented:
That being said, my end goal is freecad ifc support, and apparently this needs python bindings, so I might not *personnally* need the split.

autra commented:
I feel that I need mkDerivation (because it executes the patchElf phase) *and* buildPythonPackage. How can I combine the 2?

autra committed:
ifcopenshell: 240611 -> 0.7.10, fix build and activate most tests

By relevant release, I mean the ifcopenshell-python release, and not the blenderbim release

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.ifcopenshell</li>
    <li>python312Packages.ifcopenshell</li>
  </ul>
</details>

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.ifcopenshell</li>
    <li>python312Packages.ifcopenshell</li>
  </ul>
</details>

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.ifcopenshell</li>
    <li>python312Packages.ifcopenshell</li>
  </ul>
</details>

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.ifcopenshell</li>
    <li>python312Packages.ifcopenshell</li>
  </ul>
</details>

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.ifcopenshell</li>
    <li>python312Packages.ifcopenshell</li>
  </ul>
</details>

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.ifcopenshell</li>
    <li>python312Packages.ifcopenshell</li>
  </ul>
</details>

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.ifcopenshell</li>
    <li>python312Packages.ifcopenshell</li>
  </ul>
</details>

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

autra commented on pkgs/top-level/all-packages.nix:
```diff
@@ -8949,8 +8949,6 @@ with pkgs;
 
   idle3tools = callPackage ../tools/system/idle3tools { };
 
-  ifcopenshell = with python3Packages; toPythonApplication ifcopenshell;
```
my understanding is that this is useful if python provides some cli tools. This is not the case here. 

There *are* some binaries in `$out/bin` though. I believe that to make things properly, these should be in another derivation in libraries/, with python build disabled. Or maybe this just uses the current default python3 version and in this case I should keep this line?

autra commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5816,7 +5816,9 @@ self: super: with self; {
 
-  ifcopenshell = callPackage ../development/python-modules/ifcopenshell { };
+  ifcopenshell = callPackage ../development/python-modules/ifcopenshell {
+    inherit (pkgs) libxml2;
```
I have an error about libxml2 missing if I don't add this line, no ideas why.

SuperSandro2000 replied:
there is also a python package called libxml2

autra replied:
Ah I understand thanks, I do need the native libxml2, not the python one, that's why.

autra commented:
I believe this is now ready. I still have some questions though :-) The main one is whether or not I should keep it in "all-packages.nix"

> Tested compilation of all packages that depend on this change using nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD". Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)

For some reason, this script says "nothing to build". As the build is broken since ages, I believe we don't really need to check dependent packages ;-)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/fixing-ifcopenshell-libxml2-not-found-by-linker-at-runtime/45764/3

fehnomenal reviewed (approved):
Thanks for tackling this. I know building IfcOpenShell is not easy.

`nixpkgs-review pr 312381` also displayed "nothing to build for me" but dropped me inside a shell where I could build the derivation with `nix build ./nixpkgs#python3.pkgs.ifcopenshell`.

The executables (`IfcConvert`, `IfcGeomServer` and `svgfill`) seem to work or at least don't crash unexpectedly when provided no real input. I no longer work with IFC so I have no way to test with real data. This is also way I would like to step down as maintainer. Do you want to take over, @autra?

I'm a bit confused about the versioning. The fetched code seems to be from v0.7.10 but you reference a v0.8.0 Makefile in the comments (which is probably OK as it builds and the executables seem to run). The built `IfcConvert` also reports itself as `0.8.0`. What is going on here?

Otherwise lgtm

autra commented:
> Thanks for tackling this. I know building IfcOpenShell is not easy.

indeed 😅 

> Do you want to take over, @autra?

I can take over indeed, as I will be working with ifc files at least for a year (maybe longer).

> I'm a bit confused about the versioning. The fetched code seems to be from v0.7.10 but you reference a v0.8.0 Makefile in the comments (which is probably OK as it builds and the executables seem to run). The built IfcConvert also reports itself as 0.8.0. What is going on here?

Honestly... me too. `ifcopenshell` seems to have 3 parallel versioning systems: blenderbim (the blender plugin), ifcopenshell-python and ifcopenshell alone. As I need python binding, I use ifcopenshell-python tags, and in the last tag, the native part is already bumped as 0.8.0.

Their release process [looks very odd to me](https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.8.0/src/ifcopenshell-python/Makefile#L121). It seems that they are constructing a wheel by downloading the previous release build from s3 and copying the so and wrapper.py from there. This sounds like a very very bad practice security-wise (who is uploading to this s3 bucket?), and it's really asking for a xy-type of security issue if you ask me!

So my understanding is that they "propagate" the .so (and wrapper.py) from the latest v0.x.0 tag to each intermediate python release. If we want to mimic this behaviour, we should create a native ifcopenshell derivation built from this tag, then copy the .so from this to each python3xxPackages.ifcopenshell. The problem is that we then have a python version mismatch problem that I didn't manage to solve because the native part still needs python to build its .so and wrapper.py.

I guess we should suggest something saner upstream, but I'm not even sure what.

For now, I've decided to take everything from the python tag, but I can try again the split between native package and python packages if you have a solution to my python version mismatch issue.

What do you think about all that?

autra commented:
This issue : https://github.com/IfcOpenShell/IfcOpenShell/issues/3927 actually talks about that.

autra committed:
ifcopenshell: maintainer fehnomenal -> autra, as asked by fehnomenal

autra commented:
I fixed a nixf-tidy error and added opencollada as buildInputs.

fehnomenal commented:
Thanks for taking the time to investigate all of this and also thanks for replacing me 👍🚀

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/fixing-ifcopenshell-libxml2-not-found-by-linker-at-runtime/45764/5

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.ifcopenshell</li>
    <li>python312Packages.ifcopenshell</li>
  </ul>
</details>

autra commented:
@natsukium hi! If I'm not mistaken, you're the code owner for this. Fehnomenal already approved it and I'm now maintainer. I think it's ready to get merged. Thanks!

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4583

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,62 +1,178 @@
+    build
+    python
+    setuptools
+    wheel
```
I don't think we need wheel here

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,62 +1,178 @@
+    swig
+    # python
+    build
+    python
```
I am somewhat sure, that python is already available

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,62 +1,178 @@
+    # NOTE: it has changed a *lot* between 0.7.0 and 0.8.0, it *may* change again (look for mathutils and basically all the things this Makefile does manually)
+    substituteInPlace pyproject.toml --replace-fail "0.0.0" "${version}"
+    # NOTE: the following is directly inspired by https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.8.0/src/ifcopenshell-python/Makefile#L123
+    cp README.md ../README.bak
```
Please rm the file unless we need it for something

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,62 +1,178 @@
+    PYTHONPATH=../src/ifcopenshell-python/ python tests.py
+    popd
+    # like make test-safe, but also ignoring test/test_open.py which segfaults (this needs to be investigated)
+    pytest -p no:pytest-blender test --ignore=test/util/test_shape_builder.py --ignore=test/test_open.py
```
Can we use pytestCheckHook?

autra replied:
Just for this line or for the whole `checkPhase`? I remembered I  couldn't make it work, but as I don't remember why, I'll try again :-)

autra replied:
I've kept the `python tests.py` thing in a `preCheck`, and used `pytestCheckHook`. Can you check if the way I did it is ok? I'm not quite sure...

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,62 +1,178 @@
+  checkPhase = ''
+    cd ../../src/ifcopenshell-python
+    # installing the python wrapper and the .so, both are needed to be able to test
+    ifcopenshell_py_wrapper_path=$(find $out -name ifcopenshell_wrapper.py)
```
Can't we know the absolute/relative path?

autra replied:
It depends on the current python version (it's in `$out/lib/python-<version>` or something like that). I just found it more complicated to parse `python --version` to extract the version number and then build the path, but maybe there's a simpler way to get the current python lib folder?

Artturin replied:
> It depends on the current python version (it's in `$out/lib/python-<version>` or something like that). I just found it more complicated to parse `python --version` to extract the version number and then build the path, but maybe there's a simpler way to get the current python lib folder?

You mean `python.sitePackages`?

```
nix-repl> python3.sitePackages
"lib/python3.12/site-packages"
```

autra replied:
I've replaced these 2 finds by a path calculated from the python version.

Artturin commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,62 +1,178 @@
-    gcc10
+    # c++
     cmake
+    gcc10
```
Where is the version requirement mentioned/ Why is it needed? Please add a comment.

autra replied:
No it's not, I removed it.

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.ifcopenshell</li>
    <li>python312Packages.ifcopenshell</li>
  </ul>
</details>

autra commented:
@SuperSandro2000 @Artturin thanks for the comments! I've adressed them, please review again 😺

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,67 +1,192 @@
+    pushd ../../src/ifcopenshell-python
+    # let's test like done in .github/workflows/ci.yml
+    # installing the python wrapper and the .so, both are needed to be able to test
+    ifcopenshell_py_wrapper_path=$out/${python.sitePackages}/ifcopenshell/ifcopenshell_wrapper.py
+    cp -v $ifcopenshell_py_wrapper_path ./ifcopenshell
```
```suggestion
    cp -v $out/${python.sitePackages}/ifcopenshell/ifcopenshell_wrapper.py ./ifcopenshell
```

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,67 +1,192 @@
+    # installing the python wrapper and the .so, both are needed to be able to test
+    ifcopenshell_py_wrapper_path=$out/${python.sitePackages}/ifcopenshell/ifcopenshell_wrapper.py
+    cp -v $ifcopenshell_py_wrapper_path ./ifcopenshell
+    ifcopenshell_so_path=$out/${python.sitePackages}/ifcopenshell/_ifcopenshell_wrapper.cpython-${
+      lib.versions.major python.version + lib.versions.minor python.version
+    }-x86_64-linux-gnu.so
+    cp -v $ifcopenshell_so_path ./ifcopenshell
```
```suggestion
    cp $out/${python.sitePackages}/ifcopenshell/_ifcopenshell_wrapper.cpython-${ib.versions.major python.version + lib.versions.minor python.version}-x86_64-linux-gnu.so ./ifcopenshell
```
we probably also want to variablize the x86_64 or just use a wildcard

autra replied:
does `${system}` works in all cases here? (it would for x86_64-linux but I'm wary of differences for other plateforms)

autra replied:
Ok I went with `stdenv.targetPlatform.system`, which is necessary for cross-compilation if I understand correctlyx

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,67 +1,192 @@
+    pushd src/ifcopenshell-python
+    # The build process is here: https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.8.0/src/ifcopenshell-python/Makefile#L131
+    # NOTE: it has changed a *lot* between 0.7.0 and 0.8.0, it *may* change again (look for mathutils and basically all the things this Makefile does manually)
+    substituteInPlace pyproject.toml --replace-fail "0.0.0" "${version}"
```
This should be done in postPatch

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,67 +1,192 @@
+    # HDF5 support is currently not optional, see https://github.com/IfcOpenShell/IfcOpenShell/issues/1815
+    "-DHDF5_SUPPORT=ON"
+    "-DHDF5_INCLUDE_DIR=${hdf5.dev}/include/"
+    "-DHDF5_LIBRARIES=${hdf5.out}/lib/libhdf5_cpp.so;${hdf5.out}/lib/libhdf5.so;${zlib.out}/lib/libz.so;${libaec.out}/lib/libaec.so;" # /usr/lib64/libsz.so;"
```
```suggestion
    "-DHDF5_LIBRARIES=${lib.getLib hdf5}/lib/libhdf5_cpp.so;${lib.getLib hdf5}/lib/libhdf5.so;${lib.getLib zlib}/lib/libz.so;${lib.getLib libaec}/lib/libaec.so;"
```

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,67 +1,192 @@
     "-DOPENCOLLADA_LIBRARY_DIR=${opencollada}/lib/opencollada"
     "-DSWIG_EXECUTABLE=${swig}/bin/swig"
     "-DLIBXML2_INCLUDE_DIR=${libxml2.dev}/include/libxml2"
     "-DLIBXML2_LIBRARIES=${libxml2.out}/lib/libxml2${stdenv.hostPlatform.extensions.sharedLibrary}"
```
```suggestion
    "-DLIBXML2_LIBRARIES=${lib.getLib libxml2}/lib/libxml2${stdenv.hostPlatform.extensions.sharedLibrary}"
```

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,67 +1,192 @@
   pname = "ifcopenshell";
-  version = "240611";
+  version = "0.7.10";
   format = "other";
+  disabled = pythonOlder "3.9";
```
```suggestion
  pyproject = false;
```
we don't need to add disabled checks for old python versions

autra committed:
ifcopenshell: add a passthru.updateScript

autra commented:
Result of `nixpkgs-review pr 312381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.ifcopenshell</li>
    <li>python312Packages.ifcopenshell</li>
  </ul>
</details>

autra commented:
@SuperSandro2000 this is ready again. Because for python release, ifcopenshell doesn't use the default tag scheme, I have added an updateScript. Please review again 😃

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,67 +1,196 @@
     "-DLIBXML2_INCLUDE_DIR=${libxml2.dev}/include/libxml2"
-    "-DLIBXML2_LIBRARIES=${libxml2.out}/lib/libxml2${stdenv.hostPlatform.extensions.sharedLibrary}"
+    "-DLIBXML2_LIBRARIES=${lib.getLib libxml2}/lib/libxml2${stdenv.hostPlatform.extensions.sharedLibrary}"
+    "-DGMP_LIBRARY_DIR=${gmp.out}/lib/"
+    "-DMPFR_LIBRARY_DIR=${mpfr.out}/lib/"
```
```suggestion
    "-DGMP_LIBRARY_DIR=${lib.getLib out}/lib/"
    "-DMPFR_LIBRARY_DIR=${lib.getLib mpfr}/lib/"
```

SuperSandro2000 commented on pkgs/development/python-modules/ifcopenshell/default.nix:
```diff
@@ -1,67 +1,196 @@
+    "-DEIGEN_DIR=${eigen}/include/eigen3"
+    "-DJSON_INCLUDE_DIR=${nlohmann_json}/include/"
     "-DOCC_INCLUDE_DIR=${opencascade-occt}/include/opencascade"
     "-DOCC_LIBRARY_DIR=${opencascade-occt}/lib"
```
```suggestion
    "-DOCC_LIBRARY_DIR=${lib.getLib opencascade-occt}/lib"
```

SuperSandro2000 reviewed (approved):
LGTM

autra commented:
Thanks @SuperSandro2000, updated with your suggestion!

trofi commented:
`tests.version` needs a `package` parameter. Otherwise the eval fails as:
- https://github.com/NixOS/nixpkgs/pull/347439
